test-project/tests/compile-fail/trait-bounds-cant-coerce.rs

```diff
@@ -23,8 +23,8 @@ fn c(x: Box<Foo+Sync+Send>) {
 fn d(x: Box<Foo>) {
     a(x); //~  ERROR mismatched types
           //~| expected trait `Foo + std::marker::Send`, found trait `Foo`
-          //~| expected type `std::boxed::Box<Foo + std::marker::Send + 'static>`
-          //~| found type `std::boxed::Box<Foo + 'static>`
+          //~| expected type `std::boxed::Box<(dyn Foo + std::marker::Send + 'static)>`
+          //~| found type `std::boxed::Box<(dyn Foo + 'static)>`
```

Comment:
This fix seems simple, but I don't really understand why this is the fix?

Reply:
It's just what the latest nighty rustc outputs... they probably changed the text of the message.

Reply:
Oh, interesting... That means that even if you write `Box<Trait>` the error message is going to write `Box<dyn Trait>` (or is it `Box<(dyn Trait)>`?). I wonder if that's intentional.

Reply:
That looks strange, indeed... I assumed that here we are testing that compiletest matches errors correctly, not that rustc produces the correct message.

Reply:
To be honest, I just copied some tests from rustc's test suite when developing the test project - it was just to verify that things worked as expected, and for use as a future benchmark.

I think it's a result of the "`dyn` everywhere" approach: https://github.com/rust-lang/rust/commit/4b1808578a99b1a452b65f0bf27de4c8775e9105. In particular: https://github.com/rust-lang/rust/blob/master/src/test/ui/mismatched_types/trait-bounds-cant-coerce.stderr.